### ISSUE TITLE
Swipe-to-dismiss on sheets can sometimes dismiss the parent sheet

### ISSUE THREAD

**Issue by bhollis** [DestinyItemManager/DIM#10417]
We've had multiple reports that people who are vigorously scrolling in a scrollable sheet will see that sheet dismiss - and dismiss the parent sheet as well. The common complaint seems to be the loadout editor and the mod selection sheet - people scroll mods and it dismisses the loadout sheet entirely, losing the in-progress loadout. This is clearly a bug but it has been hard to reproduct.

Linking #9903

**Comment by ryan-rushton** [comment]
I took a stag at reproducing it in the PWA and had no lock. Any inclination of apple vs android phones and or browsers?

**Comment by bhollis** [comment]
We've had reports from both but I can't reproduce on iPhone. I bought a Samsung phone to try and see if that's easier to repro on.

**Comment by ryan-rushton** [comment]
The only think I could see as a maybe was the velocity capture point where we hide the sheet if the downward velocity is fast enough. It is possible to drag the sheet close from clicking anywhere as the drag handle is on the container. I could actually get that to trigger and close the sheet via dragging an item in it, but I couldn't replicate the behaviour reported exactly. I had to be very deliberate and it also only closed the mod sheet (I was using edit loadout + add mods).

**Comment by bhollis** [comment]
Yeah, I'm not as worried about "it's too easy to close the sheet", I'm worried about closing one sheet also closing its parent.

**Comment by ryan-rushton** [comment]
Yeah agreed

**Comment by bhollis** [comment]
My current hypothesis is that Safari buffers up scroll events while the page is blocked with updating elements, and then when it unblocks Safari dispatches those events as if they'd happened to the underlying elements.

Perhaps there's some information on the event itself that we could use to check to see if it's still relevant. Again, having a consistent repro would really help with fixing this.
